src/main/java/com/pusher/client/channel/impl/ChannelManager.java

```diff
@@ -46,6 +47,14 @@ public PrivateChannel getPrivateChannel(String channelName) throws IllegalArgume
         }
     }
 
+    public PrivateEncryptedChannel getPrivateEncryptedChannel(String channelName) throws IllegalArgumentException{
+        if (!channelName.startsWith("private-encrypted")) {
```

Comment:
We're already doing this check construction time (via [getDisallowedNameExpressions](https://github.com/pusher/pusher-websocket-java/blob/53beb30d0141f64397eea8542b2ef525af040427/src/main/java/com/pusher/client/channel/impl/PrivateEncryptedChannelImpl.java#L140)). It think it would be better to not duplicate the same logic and let the class cast exception naturally convey the same information better.

Reply:
I'm not convinced a class cast really conveys it better than a hand written message. It would also be inconsistent with the other channel types, so let's just stick to the established pattern for now.

Reply:
I do thing that the class cast exception would convey info about a programming mistake even better as you can see the actual type (i.e. maybe you should be calling `getPrivateChannel` instead). Also besides of duplication it can also lead to bugs when you can create a channel but later can't retrieve it...